src/main/java/mr/limpios/smart_divide_backend/aplication/assemblers/ExpenseModelAssembler.java

```diff
@@ -0,0 +1,126 @@
+package mr.limpios.smart_divide_backend.aplication.assemblers;
+
+import mr.limpios.smart_divide_backend.domain.models.*;
+import mr.limpios.smart_divide_backend.domain.strategies.ParticipantsPayersOfExpenses;
+import mr.limpios.smart_divide_backend.domain.strategies.ValidatedMembers;
+
+import java.math.BigDecimal;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class ExpenseModelAssembler {
+    public static List<ExpenseParticipant> createExpenseParticipantsFromValidatedParticipants(
+            ParticipantsPayersOfExpenses participantsPayersOfExpenses,
+            Map<String, User> groupMembersMap) {
+
+        Map<String, BigDecimal> payersMap = getMembersMap(participantsPayersOfExpenses.payers());
+        Map<String, BigDecimal> participantsMap = getMembersMap(participantsPayersOfExpenses.participants());
+
+        Set<String> allMemberIds = new HashSet<>();
+        allMemberIds.addAll(payersMap.keySet());
+        allMemberIds.addAll(participantsMap.keySet());
+
+        List<ExpenseParticipant> result = new ArrayList<>();
+        for (String memberId : allMemberIds) {
+            User user = groupMembersMap.get(memberId);
+            BigDecimal amountPaid = payersMap.getOrDefault(memberId, BigDecimal.ZERO);
+            BigDecimal mustPaid = participantsMap.getOrDefault(memberId, BigDecimal.ZERO);
+
+            result.add(new ExpenseParticipant(
+                    null,
+                    user,
+                    amountPaid,
+                    mustPaid
+            ));
+        }
+
+        return result;
+    }
+
+    private static Map<String, BigDecimal> getMembersMap(List<ValidatedMembers> members) {
+        return members.stream()
+                .collect(Collectors.toMap(
+                        ValidatedMembers::memberId,
+                        ValidatedMembers::amount
+                ));
+    }
```

Comment:
Crea una utilidad para esto, si no estoy mal, utilizas algo similar en otro lado. 
Puedes crear una función genérica al cuál solo se le pases el `Collection`, lo que usarás como clave y lo que quieres almacenar, 
```suggestion
   public static <T, K, V> Map<K, V> toMap(
        Collection<T> source,
        Function<T, K> keyExtractor,
        Function<T, V> valueExtractor
) {
    return source.stream()
            .collect(Collectors.toMap(keyExtractor, valueExtractor));
}
```

Y lo puedes usar así: 

`Map<String, BigDecimal> example = UtilClass.toMap(members, ValidateMembers::memberId, ValidateMembers::amount);`